### DISCUSSION TITLE
Resynch local storage on mobile with server?

### DISCUSSION THREAD

**Discussion by oblivioncth** [immich-app/immich#2463] [Q&A] [answered]
Due to some complications I had to reinstall Immich completely from scratch (after of course backing up my library). When I finally reconnected the mobile app I realized that the images must have different IDs in the database now as all of the photos that are on my phone are duplicated in the app with a non-synced and cloud only copy.

Is there any way to get Immich to recognize they're the same images?

In the end I'm fine removing them from my phone and relying on network access for those older pictures if not.

**Comment by ZhengCaiyi** [comment]
It's a P0 bug. I found this issue to. It's very serious issue, @alextran1502 this bug should be fix even if no new feature deliver.
The mobile client has very bad user experience. The sync has issue too. Almost not useable. I'm even rewriting the mobile immich app.

**Reply by samip5** [reply]
> It's a P0 bug. I found this issue to. It's very serious issue, @alextran1502 this bug should be fix even if no new feature deliver. The mobile client has very bad user experience. The sync has issue too. Almost not useable. I'm even rewriting the mobile immich app.

This is FOSS, so if you want it fixed quickly, just do so yourself and problem is solved. If you ain't paying for it, you shouldn't think it's P0 or P anything of importance. :)

Oh, and if you don't know how to fix it yourself, well good luck on the path you have chosen.

**Reply by ZhengCaiyi** [reply]
Easy, man. My experience, I backup image in 1.49, then I upgraded to 1.52, all assets became duplicated. Then I removed all data and backup again: 200G of data. And after I upgraded to 1.55, all assets became duplicated again. Then I cleared the data and backup again.
The way client check if same asset is deviceId + deviceAssetId. I believe this caused by client change the argorithom to check same asset. But I don't understand why we need to change it so frequently?
Another issue: when mobile client start or enter foreground, need to pull all assets data from server. It cost lots of data and time, expecially in LTE network, it cost mony. And the UI will become stuck when syncing assets. This is cause by  single thread mechanism of flutter.
And check the release note, we are adding more and more features into immich, but don't see people are resolving these kind of issues.
I'm working on creating a new mobile app for immich, use C++ as common sdk, and use Java for Android and Swift for iOS.
Now Android app can run and support image gallery, albums, backup. Video not supported yet. The user experience is much better than current flutter app. Depending on multi-threading, the UI is very smooth. Hoping the android app can complete all function in Aguest and open source it on github.

**Comment by alextran1502** [comment]
Hey @oblivioncth this is unexpected, did you use the same installation method for the mobile app? I.e not install from different sources? Can you layout exactly the step you took from backing up older instances. 

Usually this only happens when you switch the source of installation of the mobile app that causes the deviceID to be different

**Reply by oblivioncth** [reply]
Hi Alex, 

I actually believe the way I handled this is not something Immich can currently handle, other than the slight hope I had that it's de-deduplication mechanism might, but let me clarify to be sure:

Initial State) Server and Mobile App on 1.4x.yy (I think 1.49.0)

1. Mounted Immich's data directory via SMB and dumped all of my original images to an external system
2. Completely deleted my Immich server deployment, backing up nothing in the process (e.g. no database migration, nothing)
3. Installed a new instance of the latest version of Immich (1.55.1) and reconfigured it the same way as the old instance (same user email, same SSO, etc.)
4. Uploaded my original images to the new instance using the CLI tool
5. Updated the mobile app on my phone to the latest version on the play store
6. Logged into the server on the mobile app
7. Noted the "duplicate" entries of the same pictures, one in the "cloud only" state and the other in the "local only" state.

![Screenshot_20230518_083302](https://github.com/immich-app/immich/assets/24661585/c43823c3-92e8-43a5-ae3b-29f61e8616e1)

When first opening the app, the sync icon in the upper right shows the cross-out over the cloud like in this screenshot, but if I click on it and then return to the main screen it disappears even though nothing changed.

At no point was the mobile app deleted/re-installed, just updated as noted.

I figured that the server stores some information about sync status between itself and mobile clients with each image having a unique ID or something like that, and that because I wiped my database that information was lost so the server isn't aware that the images on my phone are the same. Again, I did this know something like that could happen, but held out hope that *maybe* Immich would somehow be smart enough to detect the images were the same anyway.

**Reply by alextran1502** [reply, marked as answer]
> Uploaded my original images to the new instance using the CLI tool

This step causes the issue you see. Because files that get uploaded from the CLI will have a different deviceId, which is not the phone's deviceId. The mobile app uses the deviceId to distinguish the remote/local asset. What should have been done was to also back up your database.

Right now the only solution is to manually change the deviceId of those records directly in the database. If you are willing to go down this road, please head over to Discord and we can help you step through the steps

**Reply by alextran1502** [reply]
A note is that we are working on clientside hashing that would eventually be the solution for such case.

**Reply by oblivioncth** [reply]
Thanks for letting me know. As I said, I was aware this likely would cause an issue. There were some complications with the TrueNAS app version of Immich that would have made database migration more annoying than usual. 

I'm no stranger to editing databases and so I imagine I can just copy the ID from one of my latest photos I've taken since after using the CLI and overwrite the one generated by the CLI client for all of the other entries. I'll probably check with the Discord just to make sure there's no other gotchyas though.

That's also really nice you plan to use hashing to make Immich's sync detection more versatile even though it's not strictly necessary.

**Reply by madadam** [reply]
> A note is that we are working on clientside hashing that would eventually be the solution for such case.

I think I have the same problem than OP. When clientside hashing lands, will the problem resolve itself or will it still require manual intervention?
